### DISCUSSION TITLE
Does async-profiler capture all the running threads every time?

### DISCUSSION THREAD

**Discussion by Thesharing** [async-profiler/async-profiler#429] [Q&A] [answered]
I fully read the README and understand that `-t` will profile threads separately. If I don't specify the `-t` option, will async-profiler capture all the running threads, or will it randomly choose a running thread and capture that thread? I'm wondering that if I could estimate the length based on the sample times.

**Comment by debashisdeb** [comment, marked as answer]
With -t, each stack trace will end with a frame that denotes a single thread.
Where as without a -t flag, the different thread samples stack traces would be merged together. So yes, it will capture all the running threads.

**Reply by apangin** [reply]
Right, `-t` affects only how samples are grouped; there is no difference in how samples are collected.

Collection of samples depends on the profiling mode:

- In `-e cpu` mode, a sample is collected whenever a thread spends the specified amount of CPU time. The more work a thread does - the more often it will be sampled. Idle threads will get no samples.
- In `-e wall` mode, all threads, including blocked or sleeping threads, will be sampled equally.
